mlir/lib/Dialect/MQTOpt/Transforms/Transpilation/Architecture.cpp

```diff
@@ -84,23 +84,69 @@ Architecture::neighboursOf(QubitIndex u) const {
   return neighbours_[u];
 }
 
-std::unique_ptr<Architecture> getArchitecture(const ArchitectureName& name) {
-  switch (name) {
-  case ArchitectureName::MQTTest: {
-    // 0 -- 1
-    // |    |
-    // 2 -- 3
-    // |    |
-    // 4 -- 5
-
+std::unique_ptr<Architecture> getArchitecture(const std::string& name) {
+  if (name == "MQTTest") {
     const Architecture::CouplingSet couplingMap{
         {0, 1}, {1, 0}, {0, 2}, {2, 0}, {1, 3}, {3, 1}, {2, 3},
         {3, 2}, {2, 4}, {4, 2}, {3, 5}, {5, 3}, {4, 5}, {5, 4}};
```

Comment:
I think you can make these coupling maps static to avoid creating them on every call.